### DISCUSSION TITLE
Is there a way to pre launch a db connection before running any files?

### DISCUSSION THREAD

**Discussion by ardenc** [vscode-abl/vscode-abl#119] [Q&A] [answered]
Is there anyway to launch a db connection using this extension and keep it open?

I ask because connecting to the db each time I run a file takes a while to make the connection before it gets to running my .p.
Ideally something like how the progress gui editor works. In that I connect to the db and it is open and connected for that session.

**Comment by gquerret** [comment]
You can create a task that will run `prowin -p _ab -db yourdb -H -S`. I think you can add ` "runOptions": { "runOn": "folderOpen" } ` to the task to start it when opening the folder.

**Comment by ardenc** [comment]
Hi, thanks so much for getting back with me.
I think this got me close ,but I might be missing something. Adding the prowin on folder open command does connect to my db and it opens up the progress appbuilder/tools.
I can open a procedure editor here and it works and is connected to my db.

However when trying to run a .p inside vs code using the extension either using ABL: Run with prowin  or ABL: Run with _progress do not share the db connection with my vs code task.

So if I have the DB connection removed in my openedge-project.json, there is no db connection so it tells me my tables etc are Unknown or ambiguous.
If I add the DB connection back it puts me back to where I was in that is takes about 15-20 seconds to begin running a .p program.

,Thanks

**Comment by gquerret** [comment, marked as answer]
VSCode always has a background session connected to the DB in order to compile the code. If you keep your own procedure editor connected to the DB, you can just execute `run xxx.p`, that will execute the code (without having to compile it as that's done by VSCode).
